src/workerd/api/tests/abortsignal-test.js

```diff
@@ -212,3 +276,273 @@ export const onabortPrototypeProperty = {
+    ok(res.counter >= 1);
+
+    // Make sure the reason was passed without being garbled
+    strictEqual(res.reason, expectedReason);
+
+    // Make sure an event was dispatched on the remote side
+    ok(res.onAbortWasFired);
+  },
+};
+
+export const rpcRequestSignal = {
+  async test(ctrl, env, ctx) {
+    // Construct a request holding an AbortSignal, and then send this request to the other side
+    const req = new Request('http://example.com', {
+      signal: AbortSignal.timeout(200),
+    });
+
+    const res = await env.RpcRemoteEnd.countToInfinityWithRequest(req);
+
+    // We don't care the exact value it got to, but at least 1 iteration should have happened
+    ok(res.counter >= 1);
+
+    // Make sure the reason was passed without being garbled
+    ok(res.reason instanceof DOMException);
+    strictEqual(res.reason.message, 'The operation was aborted due to timeout');
+
+    // Make sure an event was dispatched on the remote side
+    ok(res.onAbortWasFired);
+  },
+};
+
+export const rpcCrossRequestSignal = {
+  async test(ctrl, env, ctx) {
+    // Save an AbortController in the global scope
+    await env.RpcRemoteEnd.tryUsingGlobalAbortController();
+
+    // Try to use it again
+    await rejects(
+      async () => env.RpcRemoteEnd.tryUsingGlobalAbortController(),
+      {
+        name: 'Error',
+        message:
+          "Cannot perform I/O on behalf of a different request. I/O objects (such as streams, request/response bodies, and others) created in the context of one request handler cannot be accessed from a different request's handler. This is a limitation of Cloudflare Workers which allows us to improve overall performance. (I/O type: RefcountedCanceler)",
+      }
+    );
+  },
+};
+
+export const rpcRemoteCanIgnoreSignal = {
+  async test(ctrl, env, ctx) {
+    const ac = new AbortController();
+    const resPromise = env.RpcRemoteEnd.ignoreSignal(ac.signal);
+
+    // Wait an arbitrary amount of time, then use the AbortController to abort the remote end.
+    await scheduler.wait(200);
+    const expectedReason = 'changed my mind';
+    ac.abort(expectedReason);
+
+    const res = await resPromise;
+
+    // Every iteration completes, the remote is not reacting to the abort
+    strictEqual(res.counter, 10);
+
+    // Make sure the reason was passed without being garbled
+    strictEqual(res.reason, expectedReason);
+  },
+};
+
+export const rpcDestroySignalUnclean = {
+  async test(ctrl, env, ctx) {
+    let ac = new AbortController();
+    const resPromise = env.RpcRemoteEnd.countToInfinityWithTimeout(ac.signal);
+
+    ac.signal.skipReleaseForTest();
+    ac = null;
+    gc();
+    await scheduler.wait(10);
+
+    const res = await resPromise;
+    // This isn't what I'd expect - I can't seem to force the dropped capability logic to run when I want it
+    strictEqual(res.reason.message, 'The operation was aborted due to timeout');
+  },
+};
```

Comment:
I'm not having a good time trying to simulate the dropped capability case. This doesn't seem to actually GC the signal.